discovery/scaleway/instance.go

```diff
@@ -175,6 +177,34 @@ func (d *instanceDiscovery) refresh(ctx context.Context) ([]*targetgroup.Group,
 		}
 
 		addr := ""
+		if len(server.PublicIPs) > 0 {
+			var ipv4Addresses []string
+			var ipv6Addresses []string
+
+			for _, ip := range server.PublicIPs {
+				if ip.Family == instance.ServerIPIPFamilyInet {
+					ipv4Addresses = append(ipv4Addresses, ip.Address.String())
+				} else if ip.Family == instance.ServerIPIPFamilyInet6 {
+					ipv6Addresses = append(ipv6Addresses, ip.Address.String())
+				}
+			}
+
+			if len(ipv6Addresses) > 0 {
+				addr = ipv6Addresses[0]
```

Comment:
Thanks for the feedback
Order should be order of adding. If an IP is detached and attached again, it will be the last one in the list.
Also, the deprecated `public_ip` field will contain the first in list if it is detached.
We can also note that this chosen public ip may be an ipv6, meaning there is an issue with current code.

Reply:
We should take them as they come from the API, not sorting them

Reply:
I did not change the order. It will not change on API side unless user triggers a change.
I fixed `public_ipv4` that could be set to an ipv6.

Reply:
Thanks for the details.
`Order should be order of adding. If an IP is detached and attached again, it will be the last one in the list.`, we could add a shorter version of this to the new descriptions in `docs/configuration/configuration.md`.